### DISCUSSION TITLE
Uvicorn assumes that `HTTPResponseStartEvent.headers` is a `List`

### DISCUSSION THREAD

**Discussion by rijenkii** [Kludex/uvicorn#1779] [Potential Issue] [answered]
According to the spec and asgiref, `HTTPResponseStartEvent.headers` is `Iterable[Tuple[bytes, bytes]]`.

But, if `http.response.start` is sent like this:
```python
headers: dict
await send({
  "type": "http.response.start",
  "status": status,
  "headers": headers.items(),
})
```

Uvicorn fails with
```
  File ".../uvicorn/protocols/http/h11_impl.py", line 474, in send
    headers = self.default_headers + message_headers
              ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
TypeError: can only concatenate list (not "dict_items") to list
```

Relevant Uvicorn code: https://github.com/encode/uvicorn/blob/782f17e3855d5f36ce69d55427e74ca7f9b998c5/uvicorn/protocols/http/h11_impl.py#L471-L474

---

Potential fix may just be this:
```python
message_headers = list(message.get("headers", [])) 
```

**Comment by Kludex** [comment, marked as answer]
PR welcome.

**Reply by Kludex** [reply]
This was fixed on #1782, and it will be available on the next uvicorn release (current version is 0.20.0).
